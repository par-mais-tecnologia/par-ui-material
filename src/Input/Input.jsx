import React, { PureComponent } from 'react'
import FormControl from '@material-ui/core/FormControl/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'
import InputMUI from '@material-ui/core/Input'
import PropTypes from 'prop-types'

class Input extends PureComponent {
  state = {}

  validationInstance = this.props.validator ? this.props.validator.validator.register(
    this
  ) : false

  handleInputBlur (event) {
    if (this.props.onBlur) {
      return this.props.onBlur(event)
    } else {
      return this.validationInstance ? this.validationInstance.validate(event) : false
    }
  }

  handleInputChange (event) {
    if (this.props.onChange) {
      this.props.onChange(event)
    }
  }

  handleInputFocus (event) {
    if (this.props.onFocus) {
      this.props.onFocus(event)
    }
  }

  render () {
    const {
      helperText,
      id,
      label,
      required,
      showHelper,
      value,
      type,
      autoFocus,
      spellCheck,
      style,
      minWidth,
      maxWidth
    } = this.props

    const inputProps = {
      onBlur: this.handleInputBlur.bind(this),
      onChange: this.handleInputChange.bind(this),
      onFocus: this.handleInputFocus.bind(this),
      spellCheck: spellCheck
    }

    if (this.validationInstance) {
      this.state.errors = {
        ...this.validationInstance.error
      }
    } else {
      this.state.errors = { hasError: false }
    }

    const proxyProps = { ...this.props }
    delete proxyProps.validatorType
    delete proxyProps.showHelper
    delete proxyProps.minWidth
    delete proxyProps.maxWidth

    return (
      <FormControl
        aria-describedby={id}
        required={required}
        style={{ maxWidth, minWidth, ...style }}>
        <InputMUI
          error={this.state.errors.hasError}
          {...proxyProps}
          {...inputProps}
          label={label}
          id={id}
          margin='normal'
          value={value}
          type={type}
          autoFocus={autoFocus}
        />

        {showHelper && <FormHelperText id={id}>
          {helperText}
        </FormHelperText>}

        {this.state.errors.hasError && <FormHelperText
          id={id}> {this.state.errors.errorMessage}
        </FormHelperText>}
      </FormControl>
    )
  }
}

Input.propTypes = {
  autoFocus: PropTypes.bool,
  required: PropTypes.bool,
  spellCheck: PropTypes.bool,
  helperText: PropTypes.string,
  id: PropTypes.string,
  label: PropTypes.string,
  showHelper: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  minWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  maxWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

Input.defaultProps = {
  minWidth: '0',
  maxWidth: '100%',
  id: 'textField',
  label: '',
  required: false,
  showHelper: false,
  value: '',
  margin: 'normal',
  spellCheck: false
}

export default Input
