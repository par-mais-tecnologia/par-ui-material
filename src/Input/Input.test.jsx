import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Input from './'

configure({ adapter: new Adapter() })

export const test = describe('Input', () => {
  it('shallow Input', () => {
    const wrapper = shallow(
      <Input />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow Input match Snapshot', () => {
    const wrapper = shallow(
      <Input />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
