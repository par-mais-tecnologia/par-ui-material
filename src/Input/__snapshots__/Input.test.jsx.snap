// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input shallow Input match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    id="textField"
    label=""
    margin="normal"
    maxWidth="100%"
    minWidth="0"
    required={false}
    showHelper={false}
    spellCheck={false}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-describedby": "textField",
      "children": Array [
        <WithStyles(Input)
          error={false}
          id="textField"
          label=""
          margin="normal"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          spellCheck={false}
          value=""
        />,
        false,
        false,
      ],
      "required": false,
      "style": Object {
        "maxWidth": "100%",
        "minWidth": "0",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": undefined,
          "error": false,
          "id": "textField",
          "label": "",
          "margin": "normal",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "required": false,
          "spellCheck": false,
          "type": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "textField",
        "children": Array [
          <WithStyles(Input)
            error={false}
            id="textField"
            label=""
            margin="normal"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            spellCheck={false}
            value=""
          />,
          false,
          false,
        ],
        "required": false,
        "style": Object {
          "maxWidth": "100%",
          "minWidth": "0",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": undefined,
            "error": false,
            "id": "textField",
            "label": "",
            "margin": "normal",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "required": false,
            "spellCheck": false,
            "type": undefined,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
